### PR TITLE
Fail when end_per_testcase fails

### PR DESCRIPTION
**Pull request by arcusfelis**
Proposed changes include:
Fail when end_per_testcase fails
Use ct_groups_summary_hook from a repo MIM-1949

See tests for the hook: https://github.com/arcusfelis/ct_groups_summary_hook/pull/1

Example of failure in end_per_testcase:
```
CT results:
    0 groups passed
    0 groups failed
    0 tests eventually passed
    1 tests with end_per_testcase failed
    162 tests passed
    0 tests failed
    0 tests skipped by user
    0 tests skipped automatically

All tests done.
Build failed:
    big tests failed
Running nodes:
epmd: up and running on port 4369 with data:
name reg1 at port 51695
name fed1 at port 51631
name mongooseim3 at port 51600
name mongooseim2 at port 51598
name mongooseim at port 51405
Failed big cases:
muc_SUITE:room_registration_race_condition:create_already_registered_room
```

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235383) / elasticsearch_and_cassandra_mnesia / 01247e446ce9ffb3b35164cf6ecfac1084588987
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/235383/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/235383/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%40c5cea15a252a.2024-08-13_20.45.15/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/235418) / elasticsearch_and_cassandra_mnesia / 1e6502a58d9630a0620b2af666123bd7ad8127b0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/235418/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/235418/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4006fa508aeccb.2024-08-13_21.38.56/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

File: tools/summarise-ct-results [hunk 1]
```diff
@@ -85,9 +86,12 @@ add_up_suite_results(SuiteSummaries) ->
...
-    lists:foldl(fun(Filename, {OkAcc, FAcc, EvOkAcc}) ->
+    lists:foldl(fun(Filename, {OkAcc, FAcc, EvOkAcc, EndPerTestcaseFailedAcc}) ->
                         {ok, Terms} = file:consult(Filename),
+                        Map = maps:from_list(Terms),
                         #{eventually_ok_tests := EvOk,
-                          groups_summary := {Ok, F}} = maps:from_list(Terms),
-                        {OkAcc + Ok, FAcc + F, EvOkAcc + EvOk}
-                end, {0, 0, 0}, GroupSummaries).
+                          groups_summary := {Ok, F}}
+                            = Map,
```
Comment (chrzaszcz) [inline tools/summarise-ct-results:94]: Why two lines?

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general. I reviewed the [diff](https://github.com/arcusfelis/ct_groups_summary_hook/pull/1) and added some comments.

File: tools/summarise-ct-results (hunk 1 above)
Comment (arcusfelis) [reply tools/summarise-ct-results:94]: To have compatibility with files in the old format (where `end_per_testcase_failures` is not in the map)

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236127) / elasticsearch_and_cassandra_mnesia / d5c0a3bdb9ae4a7b6a72a1826362142c0441c373
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/236127/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/236127/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%409dce3ebec1c7.2024-08-22_23.26.31/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
Looks good

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236230) / elasticsearch_and_cassandra_mnesia / 793c6e4924eca283035bda699967fbf73de04952
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/236230/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/236230/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4055dabf456c13.2024-08-23_14.57.23/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by arcusfelis** [comment]
Looks green and now uses a hex package

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236335) / elasticsearch_and_cassandra_mnesia / 1a9e4d79c562a3f4d39990dc6be83847967a8573
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/236335/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/236335/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%401f3a7834164f.2024-08-27_16.45.22/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236571) / elasticsearch_and_cassandra_mnesia / 40010b4049dfb5be14399914d9ba8e1ba48eefa2
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/236571/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/236571/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%407d0da3c14b13.2024-08-28_08.30.28/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/236626) / elasticsearch_and_cassandra_mnesia / bde0168774dc0c83a36c402277e54b481186a3eb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4357/236626/elasticsearch_and_cassandra_mnesia.27.0.1-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4357/236626/elasticsearch_and_cassandra_mnesia.27.0.1-amd64/big/ct_run.test%4046d0e61a895d.2024-08-28_08.53.58/index.html)
**OK: 439** / Failed: 0 / **User-skipped: 47** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]

**Pull request merged**